File: spark-java-line-action/src/main/java/com/wen/spark/core/broatcast/AccumulatorDemo.java
package com.wen.spark.core.broatcast;

import org.apache.spark.Accumulator;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.VoidFunction;

import java.util.Arrays;
import java.util.List;

public class AccumulatorDemo {
    public static void main(String[] args){
        SparkConf conf=new SparkConf().setAppName("BroatCast").setMaster("local");
        JavaSparkContext sc=new JavaSparkContext(conf);
        Accumulator<Integer> sum=sc.accumulator(0);
        List<Integer> listNumber= Arrays.asList(1,2,3,4,5);
        JavaRDD<Integer> rddList=sc.parallelize(listNumber);
       rddList.foreach(new VoidFunction<Integer>(){
            public void call(Integer integer) throws Exception {
                //在java  中 获取广播变量
                sum.add(integer);
            }
    });
       System.out.println(sum.value());
    }
}

File: spark-scala-line-action/src/main/java/com/wen/spark/scala/broatcast/AccumulatorDemo.scala
package com.wen.spark.scala.broatcast

import org.apache.spark.{SparkConf, SparkContext}

object AccumulatorDemo {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setMaster("local").setAppName("AccumulatorDemo")
    val sc = new SparkContext(conf)
    val accum = sc.accumulator(0)
    sc.parallelize(Array(1, 2, 3, 4)).foreach(x => accum += x)
    println(accum)
  }
}
